src/stratis_cli/_actions/_top.py

```diff
@@ -558,3 +622,142 @@ def get_report(namespace):
+                )
+                % namespace.keydesc
+            )
+
+        # A key was updated even though there was no key reported to already exist.
+        if changed and existing_modified:  # pragma: no cover
+            raise StratisCliIncoherenceError(
+                (
+                    "Key %s was reported to not exist but stratisd reported "
+                    "that it reset an existing key"
+                )
+                % namespace.keydesc
+            )
+
+    @staticmethod
+    def reset_key(namespace):
+        """
+        Reset the key data for an existing key in the kernel keyring.
+
+        :raises StratisCliEngineError:
+        :raises StratisCliEnginePropertyError:
+        :raises StratisCliResourceNotFoundError:
+        :raises StratisCliPropertyNotFoundError:
+        :raises StratisCliIncoherenceError:
+        """
+        proxy = get_object(TOP_OBJECT)
+
+        key_list = _fetch_keylist_property(proxy)
+        if namespace.keydesc not in key_list:
+            raise StratisCliResourceNotFoundError("reset", namespace.keydesc)
+
+        ((changed, existing_modified), return_code, message) = _add_update_key(
+            proxy,
+            namespace.keydesc,
+            namespace.capture_key,
+            keyfile_path=namespace.keyfile_path,
+        )
+
+        if return_code != StratisdErrors.OK:
+            raise StratisCliEngineError(return_code, message)
+
+        # The key description existed and the key was not changed.
+        if not changed and not existing_modified:
+            raise StratisCliNoChangeError("reset", namespace.keydesc)
+
+        # A new key was added even though there was a key reported to already exist.
+        if changed and not existing_modified:  # pragma: no cover
+            raise StratisCliIncoherenceError(
+                (
+                    "Key %s was reported to already exist but stratisd reported "
+                    "that it created a new key"
+                )
+                % namespace.keydesc
+            )
+
+    @staticmethod
+    def unset_key(namespace):
+        """
+        Unset a key in kernel keyring.
+
+        :raises StratisCliEngineError:
+        :raises StratisCliEnginePropertyError:
+        :raises StratisCliPropertyNotFoundError:
+        :raises StratisCliNoChangeError:
+        :raises StratisCliIncoherenceError:
+        """
+        from ._data import Manager
+
+        proxy = get_object(TOP_OBJECT)
+
+        key_list = _fetch_keylist_property(proxy)
+        if namespace.keydesc not in key_list:
+            raise StratisCliNoChangeError("remove", namespace.keydesc)
+
+        (changed, return_code, message) = Manager.Methods.DeleteKey(
+            proxy, {"key_desc": namespace.keydesc}
+        )
+
+        if return_code != StratisdErrors.OK:  # pragma: no cover
+            raise StratisCliEngineError(return_code, message)
+
+        # A new key was added even though there was a key reported to already exist.
```

Comment:
You can drop the comment, it doesn't match the IncoherenceError message, which I think is correct.